### DISCUSSION TITLE
Add Image Capture button in addition to "upload" and "add existing"

### DISCUSSION THREAD

**Discussion by emahuni** [directus/directus#20033] [Feature Requests]
### Summary

Add a capture button that requests camera usage to capture live images as required by users.

### Basic Example

from this:

![image](https://github.com/directus/directus/assets/7290508/31bae779-e016-4178-8cbc-57a96deb348b)

to this:

![image](https://github.com/directus/directus/assets/7290508/930d2a2a-092f-437a-a3dd-51897676880d)

When capture button is pressed it shows a camera preview where one can capture the live image, previews and uploads to Directus.

### Motivation

The feature allows users to directly add images to a Directus application without first having to leave the application and save an image to disk, then return to upload. 

This allows the interface to be more useful for capturing documents (akin to scanning), user faces for avatars, on-site progress shots etc.

An in-app image capturing feature.

### Detailed Design

Basic component functionality to be merged with existing image interface:
```js
<template>
  <div>
    <div v-if="isCameraAvailable">
      <video ref="video" autoplay></video>
      <button @click="capture">Capture</button>
    </div>
    <div v-else>
      Camera not available
    </div>
    <div v-if="capturedImage">
      <img :src="capturedImage" alt="Captured" />
    </div>
  </div>
</template>

<script>
import { ref, onMounted, onUnmounted } from 'vue';

export default {
  setup() {
    const isCameraAvailable = ref(false);
    const capturedImage = ref(null);
    const videoElement = ref(null);

    const capture = () => {
      if (videoElement.value) {
        const canvas = document.createElement('canvas');
        const video = videoElement.value;
        canvas.width = video.videoWidth;
        canvas.height = video.videoHeight;
        canvas.getContext('2d').drawImage(video, 0, 0, canvas.width, canvas.height);
        capturedImage.value = canvas.toDataURL();
      }
    };

    onMounted(() => {
      const video = document.querySelector('video');
      if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
        navigator.mediaDevices.getUserMedia({ video: true })
          .then(stream => {
            isCameraAvailable.value = true;
            video.srcObject = stream;
            videoElement.value = video;
          })
          .catch(error => {
            console.error('Error accessing the camera: ', error);
          });
      } else {
        console.error('getUserMedia not supported');
      }
    });

    onUnmounted(() => {
      if (videoElement.value && videoElement.value.srcObject) {
        videoElement.value.srcObject.getTracks().forEach(track => track.stop());
      }
    });

    return {
      isCameraAvailable,
      capturedImage,
      capture,
      videoElement
    };
  }
};
</script>
```

Once the image is captured, it can then be provided for upload.

### Must Have:
- Capture button
- Captured image preview
- Captured image upload

### Should Have:
- Cropping or resizing features

### Could Have:
- Color and light correction sliders

### Won't Have:
- 

### Drawbacks

Additional code is required for image capturing, previewing and cleaning, it is simple as indicated in the detailed design. Most of the heavy lifting is being done by the media devices api.

### Alternatives

users continue to capture off-app, save and then return to app to upload images.

### Adoption Strategy

The capture feature is not a breaking change nor does it affect any existing code. It doesn't require any migrations.

### Unresolved Questions

What the design could look like to make it more user friendly, ie: not use 2 spaces for capture and preview, maybe use one space for both.

**Comment by rijkvanzanten** [comment]
I believe this is a duplicate of https://github.com/directus/directus/discussions/18997. Can you confirm/deny @emahuni? Ty! 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
